src/sentry/south_migrations/0346_backfill_email.py

```diff
@@ -0,0 +1,3289 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+class Migration(DataMigration):
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        "Write your forwards methods here."
+        # Note: Don't use "from appname.models import ModelName".
+        # Use orm.ModelName to refer to models in this application,
+        # and orm['appname.ModelName'] for models in other applications.
+        from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+        UserEmail = orm['sentry.UserEmail']
+        Email = orm['sentry.Email']
+
+        queryset = UserEmail.objects.all()
+
+        for useremail in RangeQuerySetWrapperWithProgressBar(queryset):
+            email, _ = Email.objects.get_or_create(email=useremail.email)
+            useremail.global_email = email
+            useremail.save()
```

Comment:
Don't do this `.save()` Do the `UserEmail.objects.filter(id=useremail.id).update(global_email=email)` pattern instead to avoid race condition.